fastar/core.py

```diff
@@ -65,11 +65,10 @@ def _compute_ancestral_child_counts(self, box):
         raise NotImplementedError
       ustart, ushape = ubox
       ustart = np.add(start, ustart)
-      instarts, counts, _ = dependency_rules[primitive](
+      inboxes, counts, _ = dependency_rules[primitive](
```

Comment:
Maybe we could make dependency rules more symmetrical, i. e. use the signature `rule(outbox, outcount, ...)`, where `outcount=None` indicates all ones (`Ones` can then be removed). Wdyt?